src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -187,12 +187,16 @@ def _to_chat_protocol(self, *, conversation_history, template_parameters):
                 message["context"] = m.full_response["context"]
             messages.append(message)
         template_parameters["metadata"] = {}
-        if "target_population" in template_parameters:
-            del template_parameters["target_population"]
-        if "topic" in template_parameters:
-            del template_parameters["topic"]
-        if "ch_template_placeholder" in template_parameters:
-            del template_parameters["ch_template_placeholder"]
+        for key in (
+            "conversation_starter",
+            "group_of_people",
+            "metadata",
+            "target_population",
+            "topic",
+            "ch_template_placeholder",
+        ):
+            if key in template_parameters:

```

Comment:
Nit: if using pop, you will not need the check for value:
template_parameters.pop(key, None)

Reply:
I didn't know that! fixing it

Reply:
In this case if key is not there, the default (None) will be returned.

Reply:
Done 👍🏽